### PR TITLE
[Refactoring] mailbox-store mime models

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageResultImpl.java [hunk 1]
```diff
@@ -332,9 +332,6 @@ public MimeDescriptor getMimeDescriptor() throws MailboxException {
     
     @Override
     public Headers getHeaders() throws MailboxException {
-        if (headers == null) {
-            headers = new HeadersImpl(message);
-        }
         return headers;
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageResultImpl.java]: can headers be final then?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageResultImpl.java [hunk 2]
```diff
@@ -188,27 +188,27 @@ public Content getFullContent(MimePath path) {
         return partContent.getMimeHeaders();
     }
 
-    public void setBodyContent(MimePath path, Content content) {
+    void setBodyContent(MimePath path, Content content) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageResultImpl.java]: why package-private?

**Comment by Arsnael** [comment]
```
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR] Errors: 
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsITALY:60 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsKOREA:67 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsNotITALY:81 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsNotKOREA:88 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsNotUS:74 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchHeadersTest>FetchHeaders.testFetchHeaderFieldsUS:53 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchTest>Fetch.testFetchEnvelopeIT:60 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Locat...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchTest>Fetch.testFetchEnvelopeKOREA:67 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Lo...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraFetchTest>Fetch.testFetchEnvelopeUS:53 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Locat...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameITALY:67 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Locatio...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameKOREA:60 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Locatio...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameSelectedIT:102 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] L...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameSelectedKO:109 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] L...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameSelectedUS:95 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Lo...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraRenameTest>Rename.testRenameUS:53 » InvalidServerResponse 
[c86869b1a95f87ec3bcdebaee4caed6587073495] Location: ...
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchMultipleMessagesITALY:144 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchMultipleMessagesKOREA:207 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchMultipleMessagesUS:81 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchPeekITALY:151 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchPeekKOREA:214 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchPeekUS:88 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchSingleMessageITALY:137 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchSingleMessageKOREA:200 » InvalidServerResponse
[c86869b1a95f87ec3bcdebaee4caed6587073495] [ERROR]   CassandraSelectedStateTest>SelectedState.testFetchSingleMessageUS:74 » InvalidServerResponse
```

**Comment by chibenwa** [comment]
`DiscardProtocolTest.testRecord:84 »  Timed out wait for be notified that read ...`

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**